crates/cli/src/subcommands/generate/rust.rs

```diff
@@ -937,7 +941,7 @@ impl __sdk::InModule for Reducer {{
                             for reducer in iter_reducers(module) {
                                 write!(out, "Reducer::{}", reducer_variant_name(&reducer.name));
                                 if !reducer.params_for_generate.elements.is_empty() {
-                                    // Because Rust prohibits empty struct variants,
+                                    // Because we're emitting unit variants when the payload is empty,
```

Comment:
FWIW here you can always use `{{ .. }}` to simplify the codegen logic a bit (remove the condition) if you want. Rust allows matching unit variants as if they were empty struct variants, so both `Foo::Bar => ...` and `Foo::Bar {} => ...` work fine.